prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -92,8 +93,16 @@ def login(self) -> None:
 
     def logout(self) -> None:
         """Logout from redfish."""
-        self.redfish_obj.logout()
-        logger.debug("(service) Logged out from redfish service ...")
+        # redfish sometimes fails to logout and this end up giving false alerts
+        # See https://github.com/DMTF/python-redfish-library/issues/160 and
+        # https://github.com/canonical/prometheus-hardware-exporter/issues/76
+        try:
+            self.redfish_obj.logout()
+            logger.debug("(service) Logged out from redfish service ...")
+        except BadRequestError as err:
+            logger.error("Failed to logout redfish: %s", str(err))
+            if "code: 401" not in str(err):
```

Comment:
We might consider ignoring all bad request errors during logout. It's a weird situation and nothing we can do. I don't want to trigger exception and impact the collector service.

Reply:
+1, please consider the situation that redfish is a API endpoint and sometimes you wouldn't able to connect to it due to multiple possible reasons.

Warning message should be enough in our case?